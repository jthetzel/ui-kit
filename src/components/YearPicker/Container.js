import styled from 'react-emotion';

export default styled.div`
  display: flex;
  justify-content: space-between;
  font-size: 16px;
  font-weight: bold;
  color: #263238;
  margin: 10px;
`;
