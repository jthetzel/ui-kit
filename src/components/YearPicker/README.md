YearPicker

```js
initialState = {
  date: new Date()
};

const { date } = state;

<YearPicker
  value={date.getFullYear()}
  onChange={(value) => setState({ date: new Date(date.setFullYear(value))})}
/>
```