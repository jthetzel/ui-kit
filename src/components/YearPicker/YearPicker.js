import React from 'react';
import PropTypes from 'prop-types';

// Style
import Button from './Button';
import Container from './Container';

const YearPicker = ({ value, onChange }) => (
  <Container>
    <Button onClick={onChange.bind(null, value - 1)} />
    {value}
    <Button onClick={onChange.bind(null, value + 1)} right />
  </Container>
);

YearPicker.propTypes = {
  value: PropTypes.number,
  onChange: PropTypes.func
};

YearPicker.defaultProps = {
  value: new Date().getFullYear(),
  onChange: () => {}
};

export default YearPicker;
