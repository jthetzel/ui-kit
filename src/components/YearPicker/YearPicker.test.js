import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import YearPicker from '../YearPicker';

test('YearPicker does not crash', () => {
  const wrapper = shallow(<YearPicker value={2017} onChange={() => {}} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('YearPicker change value', () => {
  const wrapper = mount(<YearPicker value={2017} onChange={() => {}} />);
  wrapper.setProps({ value: 2018 });
  expect(wrapper.props().value).toBe(2018);
});
