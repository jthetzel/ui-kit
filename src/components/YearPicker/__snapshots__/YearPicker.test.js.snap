// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearPicker does not crash 1`] = `
<Styled(div)>
  <Styled(div)
    onClick={[Function]}
  />
  2017
  <Styled(div)
    onClick={[Function]}
    right={true}
  />
</Styled(div)>
`;
